unmarshal_test.go

```diff
@@ -266,6 +266,148 @@ func TestUnmarshalOnlineQueryBulkResultDataclasses(t *testing.T) {
 	//assert.Nil(t, resultHolders[1].Dataclass)
 }
 
+func TestUnmarshalBulkQueryDataclassList(t *testing.T) {
```

Comment:
Can we also include tests with incorrect types, like trying to deserialize a string into an int, or forgetting the `*` for optional fields? The tests should check that the error messages are useful.